utils/args/GenericfMRISurfaceProcessingPipeline.py

```diff
@@ -0,0 +1,62 @@
+"""
+Builds, validates, and excecutes parameters for the HCP script 
+/opt/HCP-Pipelines/fMRISurface/GenericfMRISurfaceProcessingPipeline.sh
+part of the hcp-func gear
+"""
+import os, os.path as op
+from .common import build_command_list, exec_command
+import logging
+
+log = logging.getLogger(__name__)
+
+def build(context):
+# if [[ -z "${FW_CONFIG_RegName}" ]] || [[ $(toupper "${FW_CONFIG_RegName}") = "EMPTY" ]]; then
+#   RegName="${hcpstruct_RegName}"
+# else
+#   RegName="${FW_CONFIG_RegName}"
+# fi
+    config = context.config
+    params = {}
+    params['path'] = context.work_dir
+    params['subject'] = config['Subject']
+    params['fmriname'] = config['fMRIName']
+    #LowResMesh usually 32k vertices ("59" = 1.60mm)
+    params['lowresmesh'] = "32"
+    #****config option?****** #generally "2", "1.60" possible 
+    params['fmrires'] = "2"
+    # Smoothing during CIFTI surface and subcortical resampling
+    params['smoothingFWHM'] = params['fmrires']
+    # GrayordinatesResolution usually 2mm ("1.60" also available)
+    params['grayordinatesres'] = "2"
+    # The func gear configuration overides the struct configuration
+    # else use the struct configuration.
+    if config['RegName'] != 'Empty':
+        params['regname'] = config['RegName']
+    elif 'RegName' in context.gear_dict['hcp_struct_config']['config'].keys():
+        params['regname'] = context.gear_dict['hcp_struct_config']['config']['RegName']
+        config['RegName'] = params['regname']
+    else:
+        raise Exception('Could not set "RegName" with current configuration.')
+
+    params['printcom'] = ""
+    context.gear_dict['Surf-params'] = params
+    
+def validate(context):
+    pass 
```

Comment:
Is this the functionality intended here? Or WIP?

Reply:
It was reserved to ensure validation of the parameters....which ended up not being needed.

@dpark6060 and I have had these same conversations ;) . I will comment about its use as to not confuse.